ext/soap/php_encoding.c

```diff
@@ -2229,7 +2229,7 @@ static xmlNodePtr to_xml_array(encodeTypePtr type, zval *data, int style, xmlNod
 		return xmlParam;
 	}
 
-	if (Z_TYPE_P(data) == IS_OBJECT && instanceof_function(Z_OBJCE_P(data), zend_ce_traversable)) {
+	if (Z_TYPE_P(data) == IS_OBJECT && zend_is_iterable(data)) {
```

Comment:
Maybe it could be replaced by just `if (zend_is_iterable(data))` and moving array check (`L2290:2455`) to top of this part?

Reply:
That's not going to work, since code in this branch requires *only* an object (`zend_is_iterable` will return true for arrays, too).

Reply:
Closing this PR as performance and check were affected :(

Reply:
@tpunt true, but it would fail even if it was an array as the first part of the conditional can never be true at that point, but I do agree we should evaluate each case before changing it, alternatively we can have some macros to cover common usages like the one in question here